src/data/dfinsupp/basic.lean

```diff
@@ -59,25 +43,25 @@ section basic
 variables [Π i, has_zero (β i)] [Π i, has_zero (β₁ i)] [Π i, has_zero (β₂ i)]
 
 instance fun_like : fun_like (Π₀ i, β i) ι β :=
-⟨λ f, quotient.lift_on f pre.to_fun $ λ _ _, funext,
-  λ f g H, quotient.induction_on₂ f g (λ _ _ H, quotient.sound H) (congr_fun H)⟩
+⟨λ f, f.to_fun, λ ⟨f₁, s₁⟩ ⟨f₂, s₁⟩ (h : f₁= f₂), by { subst h, congr'} ⟩
 
 /-- Helper instance for when there are too many metavariables to apply `fun_like.has_coe_to_fun`
 directly. -/
 instance : has_coe_to_fun (Π₀ i, β i) (λ _, Π i, β i) := fun_like.has_coe_to_fun
 
+@[simp] lemma to_fun_eq_coe (f : Π₀ i, β i) : f.to_fun = f := rfl
+
 @[ext] lemma ext {f g : Π₀ i, β i} (h : ∀ i, f i = g i) : f = g := fun_like.ext _ _ h
 /-- Deprecated. Use `fun_like.ext_iff` instead. -/
 lemma ext_iff {f g : Π₀ i, β i} : f = g ↔ ∀ i, f i = g i := fun_like.ext_iff
 /-- Deprecated. Use `fun_like.coe_injective` instead. -/
 lemma coe_fn_injective : @function.injective (Π₀ i, β i) (Π i, β i) coe_fn := fun_like.coe_injective
 
-instance : has_zero (Π₀ i, β i) := ⟨⟦⟨0, ∅, λ i, or.inr rfl⟩⟧⟩
+instance : has_zero (Π₀ i, β i) := ⟨⟨0, trunc.mk $ ⟨∅, λ i, or.inr rfl⟩⟩⟩
 instance : inhabited (Π₀ i, β i) := ⟨0⟩
 
 @[simp]
-lemma coe_pre_mk (f : Π i, β i) (s : multiset ι) (hf) :
-  ⇑(⟦⟨f, s, hf⟩⟧ : Π₀ i, β i) = f := rfl
+lemma coe_mk (f : Π i, β i) (s) : ⇑(⟨f, s⟩ : Π₀ i, β i) = f := rfl
```

Comment:
You use `mk'` for the constructor name. Should the lemmas use `mk'` as well to avoid confusion with `mk` below?

Reply:
Yes, probably this should be called `coe_mk'`. Were there any others to rename?

Reply:
I did the rename. Probably I'll follow up at some point renaming `mk` to `mk_on` or something